bin/src/run/batchWarnings.js

```diff
@@ -212,13 +212,16 @@ const deferredHandlers = {
 		priority: 1,
 		fn: warnings => {
 			const nestedByPlugin = nest(warnings, 'plugin');
-
-			nestedByPlugin.forEach(({ key: plugin, items }) => {
+			nestedByPlugin.forEach(nested => {
+				const items = nested.items;
+				const plugin = nested.key;
```

Comment:
Why did you replace this destructuring?

Reply:
I did this because it was failing build on Node 4, `rollup.config.js` uses 
```js
buble({include: ['src/**', 'node_modules/acorn/**']})
```
so the same code retains destructuring in rollup.js whereas it removes in the final built file of /bin/rollup

[Build failure link](https://travis-ci.org/rollup/rollup/builds/310089193?utm_source=github_status&utm_medium=notification)

Reply:
Maybe we should change this to include the `bin` path as well. I don't see why not.

Reply:
As far as I can understand including the path in `buble` keeps the source code as is and doesn't transpile down. So if we were to include `bin` path it'll also break the build for Node 4 for destructured code as Node 4 doesn't support it.

The reason `bin/rollup` bundle works is because it gets the code correctly transpiled down and all the destructuring is removed.

Reply:
Ah, I see, I think I understand. So the problem is that now for the first time, files are imported from the `bin` folder by modules in the `src` folder.

I think the solution is simple: Common files used by both the `bin` folder and the `src` folder should got into the `src` folder. More specifically, I think that `batchWarnings.js` and `deprecateOptions` should both go into `src/utils`. This is following a pattern established for files like `relativeId.js`, `promise.js` or `ensureArray.js`. By keeping exports one way we make sure that we can always freely refactor inside the `bin` folder without possibly breaking functionality elsewhere. Then I guess you should be able to put the destructuring back in.

BTW I really like what you are doing with this PR, this duplicate deprecation was something that REALLY bugged me.